modules/caddyhttp/httpredirectlistener.go

```diff
@@ -136,18 +151,11 @@ func (c *httpRedirectConn) Read(p []byte) (int, error) {
 
 		err = resp.Write(c.Conn)
 		if err != nil {
-			errReturn = fmt.Errorf("couldn't write HTTP->HTTPS redirect")
-			return
+			return 0, fmt.Errorf("couldn't write HTTP->HTTPS redirect")
 		}
 
-		errReturn = fmt.Errorf("redirected HTTP request on HTTPS port")
-		c.Conn.Close()
```

Comment:
Why remove the conn close? The connection can no longer be used because the wrong protocol was sent to it 🤔

Reply:
The connection will be closed by the caller when encountering a non-nil error. Besides, if we close the connection ourselves, we should defer the close when it is clear the connection is sending http requests several lines above.

Reply:
Okay :man_shrugging: it's been too long since I first implemented this so I don't remember if I added `c.Conn.Close()` after testing something and noticing it's better to have, or if I just added it cause it felt like the right thing to do. I dunno. If it looks okay with `curl` and in the browser, then it should be fine I guess.